File: WellnessWingman/Models/ExerciseAnalysisResult.cs
using System.Text.Json.Serialization;

namespace HealthHelper.Models;

public class ExerciseAnalysisResult
{
    [JsonPropertyName("schemaVersion")]
    public string SchemaVersion { get; set; } = "1.0";

    [JsonPropertyName("activityType")]
    public string? ActivityType { get; set; }

    [JsonPropertyName("metrics")]
    public ExerciseMetrics Metrics { get; set; } = new();

    [JsonPropertyName("insights")]
    public ExerciseInsights? Insights { get; set; }

    [JsonPropertyName("warnings")]
    public List<string> Warnings { get; set; } = new();
}

public class ExerciseMetrics
{
    [JsonPropertyName("distance")]
    public double? Distance { get; set; }

    [JsonPropertyName("distanceUnit")]
    public string? DistanceUnit { get; set; }

    [JsonPropertyName("durationMinutes")]
    public double? DurationMinutes { get; set; }

    [JsonPropertyName("averagePace")]
    public string? AveragePace { get; set; }

    [JsonPropertyName("averageSpeed")]
    public double? AverageSpeed { get; set; }

    [JsonPropertyName("speedUnit")]
    public string? SpeedUnit { get; set; }

    [JsonPropertyName("calories")]
    public double? Calories { get; set; }

    [JsonPropertyName("averageHeartRate")]
    public double? AverageHeartRate { get; set; }

    [JsonPropertyName("maxHeartRate")]
    public double? MaxHeartRate { get; set; }

    [JsonPropertyName("steps")]
    public int? Steps { get; set; }

    [JsonPropertyName("elevationGain")]
    public double? ElevationGain { get; set; }

    [JsonPropertyName("elevationUnit")]
    public string? ElevationUnit { get; set; }
}

public class ExerciseInsights
{
    [JsonPropertyName("summary")]
    public string? Summary { get; set; }

    [JsonPropertyName("positives")]
    public List<string> Positives { get; set; } = new();

    [JsonPropertyName("improvements")]
    public List<string> Improvements { get; set; } = new();

    [JsonPropertyName("recommendations")]
    public List<string> Recommendations { get; set; } = new();
}

File: shared/src/commonMain/kotlin/com/wellnesswingman/data/model/analysis/ExerciseAnalysisResult.kt
package com.wellnesswingman.data.model.analysis

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable

/**
 * Structured JSON schema for exercise analysis results from the LLM.
 */
@Serializable
data class ExerciseAnalysisResult(
    @SerialName("schemaVersion")
    val schemaVersion: String = "1.0",

    @SerialName("activityType")
    val activityType: String? = null,

    @SerialName("metrics")
    val metrics: ExerciseMetrics = ExerciseMetrics(),

    @SerialName("insights")
    val insights: ExerciseInsights? = null,

    @SerialName("warnings")
    val warnings: List<String> = emptyList()
)

@Serializable
data class ExerciseMetrics(
    @SerialName("distance")
    @Serializable(with = LenientNullableDoubleSerializer::class)
    val distance: Double? = null,

    @SerialName("distanceUnit")
    val distanceUnit: String? = null,

    @SerialName("durationMinutes")
    @Serializable(with = LenientNullableDoubleSerializer::class)
    val durationMinutes: Double? = null,

    @SerialName("averagePace")
    val averagePace: String? = null,

    @SerialName("averageSpeed")
    @Serializable(with = LenientNullableDoubleSerializer::class)
    val averageSpeed: Double? = null,

    @SerialName("speedUnit")
    val speedUnit: String? = null,

    @SerialName("calories")
    @Serializable(with = LenientNullableDoubleSerializer::class)
    val calories: Double? = null,

    @SerialName("averageHeartRate")
    @Serializable(with = LenientNullableDoubleSerializer::class)
    val averageHeartRate: Double? = null,

    @SerialName("maxHeartRate")
    @Serializable(with = LenientNullableDoubleSerializer::class)
    val maxHeartRate: Double? = null,

    @SerialName("steps")
    @Serializable(with = LenientNullableDoubleSerializer::class)
    val steps: Double? = null,

    @SerialName("elevationGain")
    @Serializable(with = LenientNullableDoubleSerializer::class)
    val elevationGain: Double? = null,

    @SerialName("elevationUnit")
    val elevationUnit: String? = null
)

@Serializable
data class ExerciseInsights(
    @SerialName("summary")
    val summary: String? = null,

    @SerialName("positives")
    val positives: List<String> = emptyList(),

    @SerialName("improvements")
    val improvements: List<String> = emptyList(),

    @SerialName("recommendations")
    val recommendations: List<String> = emptyList()
)
